modules/security/pam.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.security.pam;
+
+  # Implementation Notes
+  #
+  # We don't use `environment.etc` because this would require that the user manually delete
+  # `/etc/pam.d/sudo` which seems unwise given that applying the nix-darwin configuration requires
+  # sudo. We also can't use `system.patchs` since it only runs once, and so won't patch in the
+  # changes again after OS updates (which remove modifications to this file).
```

Comment:
What's this about `system.patches`? I haven't used it before but reading through its activation script the logic appears to be "apply the patch unless it's already applied" (where "already applied" is defined as "it can be successfully reversed"). The part that looks up the previously-applied patches is just to reverse any patches that have been removed.

Which is to say, it certainly appears as though it will attempt to repatch on every activation.

Reply:
It turns out, `system.patches` is actually broken. This PR should fix it: https://github.com/LnL7/nix-darwin/pull/612

Reply:
With #612 applied to my local setup, I am currently using this configuration to enable touch ID:

```
  # Enable Enable sudo authentication with Touch ID
  system.patches = [
    (pkgs.writeText "pam.patch" ''
      --- a/etc/pam.d/sudo
      +++ b/etc/pam.d/sudo
      @@ -1,4 +1,6 @@
       # sudo: auth account password session
      +auth       optional       ${pkgs.pam-reattach}/lib/pam/pam_reattach.so  # Needed for using Touch ID within tmux
      +auth       sufficient     pam_tid.so
       auth       sufficient     pam_smartcard.so
       auth       required       pam_opendirectory.so
       account    required       pam_permit.so
    '')
  ];
```

This works pretty well and gets applied on every update. It is quite resilient and will continue to work even if I make changes to the patch (ex. changing the tmux related comment). This is because `system.patches` will reverse outdated patches before applying new ones.